test/presenters/return-submissions/view-return-submission.presenter.test.js

```diff
@@ -0,0 +1,275 @@
+
+          expect(result.tableData.cubicMetresTotal).to.equal('28,000')
+          expect(result.tableData.unitTotal).to.equal('28,000')
+        })
+      })
+
+      describe('and the volumes are not cubic metres', () => {
+        beforeEach(() => {
+          testReturnSubmission = createSubmission({ userUnit: unitNames.GALLONS })
+          Sinon.stub(testReturnSubmission, '$units').returns(unitNames.GALLONS)
+        })
+
+        it('includes the expected headers', () => {
+          const result = ViewReturnSubmissionPresenter.go(testReturnSubmission, '2025-1')
+
+          const headers = result.tableData.headers.map((header) => header.text)
+
+          expect(headers).to.equal(['Day', 'Gallons', 'Cubic metres'])
+        })
+
+        it('includes the expected rows', () => {
+          const result = ViewReturnSubmissionPresenter.go(testReturnSubmission, '2025-1')
+
+          expect(result.tableData.rows.length).to.equal(28)
+          // We use include() as a row can also include a reading key which we don't care about for volumes
+          expect(result.tableData.rows[0]).to.include({
+            cubicMetresQuantity: '219,969.248',
+            date: '1 February 2025',
+            unitQuantity: '1,000'
+          })
+        })
+
+        it('includes the expected totals', () => {
+          const result = ViewReturnSubmissionPresenter.go(testReturnSubmission, '2025-1')
+
+          expect(result.tableData.cubicMetresTotal).to.equal('6,159,138.952')
+          expect(result.tableData.unitTotal).to.equal('28,000')
+        })
+      })
+    })
+
+    describe('when the return submission contains readings', () => {
+      beforeEach(() => {
+        testReturnSubmission = createSubmission({ readings: true })
+        Sinon.stub(testReturnSubmission, '$method').returns('oneMeter')
+      })
+
+      it('includes the expected headers', () => {
+        const result = ViewReturnSubmissionPresenter.go(testReturnSubmission, '2025-1')
+
+        const headers = result.tableData.headers.map((header) => header.text)
+
+        expect(headers).to.equal(['Day', 'Reading', 'Cubic metres'])
+      })
+
+      it('includes the expected rows', () => {
+        const result = ViewReturnSubmissionPresenter.go(testReturnSubmission, '2025-1')
+
+        expect(result.tableData.rows.length).to.equal(28)
+        // We use include() as a row can also include a reading key which we don't care about for volumes
+        expect(result.tableData.rows[0]).to.include({
+          reading: 1001,
+          date: '1 February 2025'
+        })
+      })
+
+      it('includes the expected totals', () => {
+        const result = ViewReturnSubmissionPresenter.go(testReturnSubmission, '2025-1')
+
+        expect(result.tableData.cubicMetresTotal).to.equal('28,000')
+      })
+    })
+  })
+})
+
+function createSubmission({ userUnit = unitNames.CUBIC_METRES, readings = false } = {}) {
```

Comment:
Private functions should be prefixed with an underscore, e.g. `_createSubmission`